sprout/appliances/models.py

```diff
@@ -671,9 +671,24 @@ def prolong_lease(self, time=60):
             appliance.prolong_lease(time=time)
 
     def kill(self):
+        save_lives = not self.fulfilled
         if self.appliances:
             for appliance in self.appliances:
-                Appliance.kill(appliance)
+                # The leased_until is reliable sign of whether the appliance was used
+                # Unless someone was messing with DB ;)
+                if appliance.ready and appliance.leased_until is None and save_lives:
```

Comment:
Since `save_lives` is already calculated, it would save a miniscule amount of effort to check that before `appliance.ready`/`.leased_until`.